pkg/apis/common/v1/common.go

```diff
@@ -120,6 +120,7 @@ type ObjectSelector struct {
 	// - `username`: the username of the user to be authenticated to the Elastic resource
 	// - `password`: the password of the user to be authenticated to the Elastic resource
 	// - `ca.crt`: the CA certificate in PEM format (optional).
+	// - `apikey`: the key to authenticate against the Elastic resource instead of a username and password (supported only for `elasticsearchRefs` in AgentSpec).
```

Comment:
?
```suggestion
	// - `api-key`: the key to authenticate against the Elastic resource instead of a username and password (supported only for `elasticsearchRefs` in AgentSpec).
```